crates/agent/src/connector_tags.rs

```diff
@@ -96,57 +98,171 @@ impl TagHandler {
             }
         }
 
-        // Fetch its connector specification.
-        let spec = jobs::run_with_output(
-            "spec",
-            &self.logs_tx,
-            row.logs_token,
-            async_process::Command::new(format!("{}/flowctl-go", &self.bindir))
-                .arg("api")
-                .arg("spec")
-                .arg("--image")
-                .arg(&image_composed)
-                .arg("--network")
-                .arg(&self.connector_network),
-        )
-        .await?;
+        let proto_type = match runtime::flow_runtime_protocol(&image_composed).await {
```

Comment:
FYI no action -- we've not really talked about it IIRC -- but I've been thinking / endeavoring to keep the protobuf tags and names equal so that Spec request / response can be a "universal" RPC that's applied without knowing its type a priori.

This is still pretty half baked. I think it would mean, for example, arbitrarily picking `Runtime::unary_capture` to get a spec response even though you have no idea what the actual connector type is.

(And Derivation connectors do also have a Spec RPC)

Reply:
yeah, and this code makes the problem pretty apparent. In the case of _image_ connectors, we can know the protocol up front by inspecting the image. But for other (local) connectors, there's no way to tell ahead of time unless we store the configuration elsewhere.

I think it would be nice if we could come up with a way to make `spec` a "universal" RPC. Another idea would be to pull the `protocol` from the `connector_tags` table instead of the other way around. Basically, make whoever deploys the connector specify what type of connector it is.